utils-intervalrdd/src/main/scala/org/bdgenomics/utils/interval/array/IntervalArray.scala

```diff
@@ -144,6 +145,36 @@ trait IntervalArray[K <: Interval[K], T] extends Serializable {
     }
   }
 
+  /**
+   * Finds an index in array that overlaps with the query rr, or if none exists
```

Comment:
in docs `rr` is also referred to as an interval, a range, a region, and K.  Would be nice if these were consistent.  I'd vote for `interval`, since that is the name of the class.

If you are interested in doing some doc cleanup in this pull request, `(K,V)` in the docs should be `(K,T)`, and all doc comment strings should be full sentences.  Otherwise I can do that separately.